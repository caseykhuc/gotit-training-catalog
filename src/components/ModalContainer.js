import React from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import modalKeys from 'constants/modelKeys';
import RegisterModal from 'components/Modal/RegisterModal';
import SignInModal from 'components/Modal/SignInModal';
import DeleteItemModal from 'components/Modal/DeleteItemModal';
import EditItemModal from 'components/Modal/EditItemModal'
import AddItemModal from 'components/Modal/AddItemModal';

export const ModalContainer = ({ modal }) => {
  const renderModal = ({ current, props }) => {
    switch (current) {
      case modalKeys.REGISTER_MODAL:
        return <RegisterModal />;
      case modalKeys.SIGN_IN_MODAL:
        return <SignInModal />;
      case modalKeys.DELETE_ITEM_MODAL:
        return <DeleteItemModal {...props} />;
      case modalKeys.EDIT_ITEM_MODAL:
        return <EditItemModal {...props} />;
      case modalKeys.ADD_ITEM_MODAL:
        return <AddItemModal {...props} />
      default:
        return '';
    }
  };

  // return <div>ModalContainer {modal.current}</div>;
  return (
    <div style={{
      position: 'absolute',
      top: '50%',
      left: '50%',
      transform: 'translate(-50%, -50%)',
    }}
    >
      {renderModal(modal)}
    </div>
  );
};

ModalContainer.propTypes = {
  modal: PropTypes.shape({
    current: PropTypes.string,
  }).isRequired,
}

const mapStateToProps = (state) => ({
  modal: state.modal,
});

export default connect(mapStateToProps)(ModalContainer);
