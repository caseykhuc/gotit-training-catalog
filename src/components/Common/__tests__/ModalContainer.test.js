import React from 'react';
import { shallow } from 'enzyme';
import modalKeys from 'constants/modelKeys';
import { ModalContainer } from '../ModalContainer';

describe('component/Common/ModalContainer', () => {
  let props;
  let wrapper;

  const update = () => {
    wrapper.update();
  };
  const setup = () => {
    wrapper = shallow(<ModalContainer {...props} />);
    update();
  };
  beforeEach(() => {
    props = {
      modal: {
        current: 'SAMPLE_MODAL',
      },
    }
  });
  it('should render correctly', () => {
    setup();
    expect(wrapper).toMatchSnapshot();

    Object.keys(modalKeys).map((key) => {
      props.modal.current = modalKeys[key];
      setup();
      return expect(wrapper).toMatchSnapshot();
    });
  });
})
