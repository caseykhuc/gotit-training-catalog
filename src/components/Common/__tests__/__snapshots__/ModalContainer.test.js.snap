// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component/Common/ModalContainer should render correctly 1`] = `ShallowWrapper {}`;

exports[`component/Common/ModalContainer should render correctly 2`] = `ShallowWrapper {}`;

exports[`component/Common/ModalContainer should render correctly 3`] = `ShallowWrapper {}`;

exports[`component/Common/ModalContainer should render correctly 4`] = `ShallowWrapper {}`;

exports[`component/Common/ModalContainer should render correctly 5`] = `ShallowWrapper {}`;

exports[`component/Common/ModalContainer should render correctly 6`] = `ShallowWrapper {}`;
